modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -478,64 +525,122 @@ private List<ObjectValidationRule> _getObjectValidationRules(
+			objectDefinition.isApproved(),
+			objectValidationRule.getObjectValidationRuleId(),
+			ObjectValidationRuleSettingConstants.NAME_OUTPUT_OBJECT_FIELD_ID,
+			objectValidationRuleSettingValuesMap.get(
+				ObjectValidationRuleSettingConstants.
+					NAME_OUTPUT_OBJECT_FIELD_ID));
 
-			if (delete) {
-				_objectValidationRuleSettingPersistence.remove(
-					oldObjectValidationRuleSetting);
-			}
-		}
+		List<ObjectValidationRuleSetting> updatedObjectValidationRuleSettings =
+			new ArrayList<>();
 
 		for (ObjectValidationRuleSetting objectValidationRuleSetting :
 				objectValidationRuleSettings) {
 
-			_objectValidationRuleSettingLocalService.
-				addObjectValidationRuleSetting(
-					objectValidationRule.getUserId(),
+			ObjectValidationRuleSetting oldObjectValidationRuleSetting =
+				_objectValidationRuleSettingPersistence.fetchByOVRI_N_V(
 					objectValidationRule.getObjectValidationRuleId(),
 					objectValidationRuleSetting.getName(),
 					objectValidationRuleSetting.getValue());
+
+			if (oldObjectValidationRuleSetting != null) {
+				updatedObjectValidationRuleSettings.add(
+					oldObjectValidationRuleSetting);
+
+				continue;
+			}
+
+			if (!objectDefinition.isApproved() ||
+				!objectValidationRuleSetting.compareName(
+					ObjectValidationRuleSettingConstants.
+						NAME_COMPOSITE_KEY_OBJECT_FIELD_ID)) {
+
+				continue;
+			}
+
+			ObjectFieldLocalService objectFieldLocalService =
+				_objectFieldLocalServiceSnapshot.get();
+
+			ObjectField objectField = objectFieldLocalService.getObjectField(
+				GetterUtil.getLong(objectValidationRuleSetting.getValue()));
+
+			Table<?> table = objectFieldLocalService.getTable(
+				objectValidationRule.getObjectDefinitionId(),
+				objectField.getName());
+
+			Column<?, ?> column = table.getColumn(
+				objectField.getDBColumnName());
+
+			ObjectEntryLocalService objectEntryLocalService =
+				_objectEntryLocalServiceSnapshot.get();
+
+			long objectEntriesCount =
+				objectEntryLocalService.getObjectEntriesCount(
+					0, objectDefinition, column.isNotNull());
+
+			if (objectEntriesCount > 0) {
+				throw new ObjectValidationRuleSettingValueException.
+					InvalidValue(
+						objectValidationRuleSetting.getName(),
+						objectValidationRuleSetting.getValue());
+			}
+
+			updatedObjectValidationRuleSettings.add(
+				_objectValidationRuleSettingLocalService.
+					addObjectValidationRuleSetting(
+						objectValidationRule.getUserId(),
+						objectValidationRule.getObjectValidationRuleId(),
+						objectValidationRuleSetting.getName(),
+						objectValidationRuleSetting.getValue()));
 		}
 
-		return newObjectValidationRuleSettings;
+		return updatedObjectValidationRuleSettings;
 	}
 
 	private void _validate(
 			String externalReferenceCode, long objectValidationRuleId,
-			long objectDefinitionId, long companyId, String engine,
-			Map<Locale, String> nameMap, String outputType, String script,
+			boolean active, long objectDefinitionId, long companyId,
```

Comment:
@mateussgn please follow the same order from the [service.xml](https://github.com/liferay/liferay-portal/blob/a839674f12b37039198175bdd9a592cd531d1d0d/modules/apps/object/object-service/service.xml#L771-L788)